core/src/test/scala/unit/kafka/log/LogCleanerIntegrationTest.scala

```diff
@@ -42,6 +42,11 @@ class LogCleanerIntegrationTest extends AbstractLogCleanerIntegrationTest with K
   val time = new MockTime()
   val topicPartitions = Array(new TopicPartition("log", 0), new TopicPartition("log", 1), new TopicPartition("log", 2))
 
+  @After
+  def cleanup(): Unit = {
+    TestUtils.clearYammerMetrics()
```

Comment:
Can you clarify how this fixes the problem?

Reply:
I've seen on the JIRA description that the test failed on the assertion of the metric so I thought the cleanup process can be incorrect, therefore I changed it to clean it up properly after each test.
If you think this isn't a reasonable basis for this test failure I can dig deeper as soon as I find another occurrence in one of the currently running Jenkins builds, maybe that contains more information (since the ones on the jira were already deleted).

Reply:
I think the cleanup logic was indeed broken because it assumed that the metric already existed. I thought another possibility is that we were finding the fetcher DeadThreadCount metric. I think we could make this more resilient if we use `getGauge(name, scope)` to make sure we are finding the right metric.

Reply:
True. I added filtering for scope too but with the twist that I check for null scope instead as the LogCleaner one doesn't have any :)